src/main/java/spoon/support/reflect/reference/CtTypeReferenceImpl.java

```diff
@@ -357,24 +359,43 @@ public void replace(CtTypeReference<?> reference) {
 		Collection<CtFieldReference<?>> l = new ArrayList<>();
 		CtType<?> t = getDeclaration();
 		if (t == null) {
-			for (Field f : getActualClass().getDeclaredFields()) {
-				l.add(getFactory().Field().createReference(f));
-			}
-			if (getActualClass().isAnnotation()) {
-				for (Method m : getActualClass().getDeclaredMethods()) {
-					CtTypeReference<?> retRef = getFactory().Type().createReference(m.getReturnType());
-					CtFieldReference<?> fr = getFactory().Field().createReference(this, retRef, m.getName());
-					// fr.
-					l.add(fr);
-				}
+			try {
+				createFieldReferences(l);
+			} catch (SpoonClassNotFoundException cnfe) {
+				return handleParentNotFound(cnfe);
 			}
-
 		} else {
 			return t.getDeclaredFields();
 		}
 		return l;
 	}
 
+	private void createFieldReferences(Collection<CtFieldReference<?>> l) {
```

Comment:
some improvements:
- rename it in `getDeclaredFieldReferences`
- don't take `l` as parameter, create it inside and return it.
- add a one line javadoc